fabric/patches/2_0001-Extension-point-for-go-build-tag-enabled-additional-.patch

```diff
@@ -0,0 +1,79 @@
+From 7abcf2a461eb6f6289d797d0142de4f7e1fdb5ff Mon Sep 17 00:00:00 2001
+From: michael steiner <michael.steiner@intel.com>
+Date: Fri, 31 Jul 2020 14:21:19 -0700
+Subject: [PATCH 1/2] Extension point for go build-tag enabled additional SCCs
+
+Signed-off-by: michael steiner <michael.steiner@intel.com>
+---
+ internal/peer/node/sccs_opt.go | 31 +++++++++++++++++++++++++++++++
+ internal/peer/node/start.go    |  9 ++++++++-
+ 2 files changed, 39 insertions(+), 1 deletion(-)
+ create mode 100644 internal/peer/node/sccs_opt.go
+
+diff --git a/internal/peer/node/sccs_opt.go b/internal/peer/node/sccs_opt.go
+new file mode 100644
+index 000000000..d46fa4811
+--- /dev/null
++++ b/internal/peer/node/sccs_opt.go
+@@ -0,0 +1,31 @@
++/*
++ Copyright 2020 Intel Corporation
++
++SPDX-License-Identifier: Apache-2.0
++*/
++
++package node
++
++import (
++	"github.com/hyperledger/fabric/core/aclmgmt"
++	"github.com/hyperledger/fabric/core/peer"
++	"github.com/hyperledger/fabric/core/scc"
++)
++
++type optionalSCCFactory interface {
++	newOptSCC(aclProvider aclmgmt.ACLProvider, p *peer.Peer) scc.SelfDescribingSysCC
++}
++
++type optionalSCCRegistry struct {
++	registeredFactories []optionalSCCFactory
++}
++
++func (f *optionalSCCRegistry) add(sf optionalSCCFactory) {
++	f.registeredFactories = append(f.registeredFactories, sf)
++}
++
++func (f *optionalSCCRegistry) list() []optionalSCCFactory {
++	return f.registeredFactories
++}
++
++var optionalSCCs = optionalSCCRegistry{registeredFactories: make([]optionalSCCFactory, 0)}
+diff --git a/internal/peer/node/start.go b/internal/peer/node/start.go
+index 63c0ae919..22557bd98 100644
+--- a/internal/peer/node/start.go
++++ b/internal/peer/node/start.go
+@@ -595,6 +595,13 @@ func serve(args []string) error {
+ 		"_lifecycle": {},
+ 	}
+ 
++	optSCCInsts := []scc.SelfDescribingSysCC{}
++	for _, sf := range optionalSCCs.list() {
++		osi := sf.newOptSCC(aclProvider, peerInstance)
++		optSCCInsts = append(optSCCInsts, osi)
++		builtinSCCs[osi.Name()] = struct{}{}
++	}
```

Comment:
As we want to have this upstream at some point we should also have some tests covering loading more than one optSSC.

Reply:
i guess for test-cases having zero, one or more (say 3) optSSC should be sufficient.  Probably should also test that the handler returns non-nil and then do a corresponding test-case.  Any other error-condition i should check (and write test-case for it)?

Reply:
ah this test looks already good to me. No other error condition comes to my mind at the moment.
As a testing bonus maybe one could use ginkgo/gomega to make tests shorter (and more readable) as often used in Fabric. Here an [example](https://github.com/hyperledger/fabric/blob/master/core/endorser/endorser_test.go).

So instead of 
```go
ff = opt_scc.ListFactoryFuncs()
if ff == nil {
    t.Error("factory function list is nil")
}
if len(ff) != 3 { ...
```
you could write something like that
```go
import . "github.com/onsi/gomega"

ff = opt_scc.ListFactoryFuncs()
Expect(ff).ShouldNot(BeNil())
Expect(len(ff)).Should(Equal(3))
```

Reply:
gingko seemed overkill by maybe just gomega might help.  Seems gomega also supports `reflect.DeepEqual`. Let me give a try ...

Reply:
doesn't work as outlined above: gomega requires a registered fail handler and the only pre-existing one i saw is the one with ginko. And if you register that but do not use ginko for the rest, then you don't get meaningful output if your test fails and actually get a stack trace. I.e., you would have to do the whole ginko shabang which i don't think is worth the effort (nor personally do i think it is really more readable).  I.e., i'll leave if for now ...

Reply:
actually, what _does_ work is to do it xunit style. slightly different syntax but somewhat same semantics ..